core/src/main/java/io/aiven/klaw/service/utils/CacheService.java

```diff
@@ -74,17 +74,11 @@ public Optional<T> get(int tenantId, Integer id) {
   }
 
   public Map<Integer, T> getCache(Integer tenantId) {
-    if (!cache.containsKey(tenantId)) {
-      cache.put(tenantId, new HashMap<>());
-    }
-    return cache.get(tenantId);
+    return cache.computeIfAbsent(tenantId, k -> new HashMap<>());
```

Comment:
I have to do a small bit of testing on this but it may get delayed until the next release is out I am currently testing the release candidate.

Reply:
Back reviewing this now after release, sorry for the delay!